Assets/MixedRealityToolkit.SDK/Experimental/Features/Dwell/DwellHandler.cs

```diff
@@ -0,0 +1,243 @@
+{
+    [Serializable]
+    public class PointerUnityEvent : UnityEvent<IMixedRealityPointer>
+    {
+    }
+
+    /// <summary>
+    /// Use this component to add a Dwell modality (https://docs.microsoft.com/en-us/windows/mixed-reality/gaze-and-dwell) to the UI target.
+    /// </summary>
+    public class DwellHandler : MonoBehaviour, IMixedRealityFocusHandler
+    {
+        protected enum DwellStateType
+        {
+            None = 0,
+            FocusGained,
+            DwellIntended,
+            DwellStarted,
+            DwellCompleted,
+            DwellCanceled,
+        }
+
+        [Header("Dwell Settings")]
+        [SerializeField]
+        protected DwellProfile dwellProfile = null;
+
+        [Header("Dwell Events")]
+        [SerializeField]
+        private PointerUnityEvent DwellIntended = new PointerUnityEvent();
+
+        [SerializeField]
+        private PointerUnityEvent DwellStarted = new PointerUnityEvent();
+
+        [SerializeField]
+        private PointerUnityEvent DwellCompleted = new PointerUnityEvent();
+
+        [SerializeField]
+        private PointerUnityEvent DwellCanceled = new PointerUnityEvent();
+
+        /// <summary>
+        /// Invoke this event if the dwell is interrupted with alternate modality
+        /// </summary>
+        public UnityAction CancelDwellAction;
+
+        /// <summary>
+        /// Property exposing the computation for what percentage of dwell has progressed.
+        /// </summary>
+        public virtual float DwellProgress
+        {
+            get
+            {
+                switch (CurrentDwellState)
+                {
+                    case DwellStateType.None:
+                    case DwellStateType.FocusGained:
+                        return 0;
+                    case DwellStateType.DwellStarted:
+                        return Mathf.Clamp((float)(dwellProfile.TimeToCompleteDwell - FillTimer) / dwellProfile.TimeToCompleteDwell, 0f, 1f);
+                    case DwellStateType.DwellCompleted:
+                        return 1;
+                    case DwellStateType.DwellCanceled:
+                        if ((DateTime.UtcNow - focusExitTime).TotalSeconds > dwellProfile.TimeToAllowDwellResume)
+                        {
+                            CurrentDwellState = DwellStateType.None;
+                            return 0;
+                        }
+                        else if (dwellProfile.TimeToAllowDwellResume > 0)
+                        {
+                            return Mathf.Clamp((float)(dwellProfile.TimeToCompleteDwell - FillTimer) / dwellProfile.TimeToCompleteDwell, 0f, 1f);
+                        }
+                        break;
+                    default:
+                        return 0;
+                }
+
+                return 0;
+            }
+        }
+
+        /// <summary>
+        /// Cached pointer reference to track focus events maps to teh same pointer that initiated dwell
```

Comment:
>Cached pointer reference to track focus events maps to teh same pointer that initiated dwell [](start = 12, length = 92)

Cached pointer reference to track focus events maps to the same pointer that initiated the dwell.